packages/gatsby-image/src/index.js

```diff
@@ -37,18 +37,17 @@ const convertProps = props => {
     convertedProps.loading = `eager`
   }
 
+  // convert fluid & fixed to arrays so we only have to work with arrays
+  convertedProps.fluid = [].concat(convertedProps.fluid).filter(Boolean)
+  convertedProps.fixed = [].concat(convertedProps.fixed).filter(Boolean)
+
   return convertedProps
 }
 
 // Find the source of an image to use as a key in the image cache.
-// Use `fixed` or `fluid` if specified, or the first image in
-// either `fixedImages` or `fluidImages`
-const getImageSrcKey = ({ fluid, fixed, fluidImages, fixedImages }) => {
-  const data =
-    fluid ||
-    fixed ||
-    (fluidImages && fluidImages[0]) ||
-    (fixedImages && fixedImages[0])
+// Use `the first image in either `fixed` or `fluid`
+const getImageSrcKey = ({ fluid, fixed }) => {
+  const data = (fluid.length && fluid[0]) || (fixed.length && fixed[0])
```

Comment:
Is the length check necessary? It'll likely fail iirc checking the `length` property if the prop itself is undefined? Or is this because of `convertProps()` that it's safe to always assume an array value present?

I would add that we have both `inImageCache()` and `activateCacheForImage()` both calling this method after first doing:

```JS
const convertedProps = convertProps(props)
```

We might as well just move that duplication into this method and drop the need to pass in props?